wrapanapi/systems/nuage.py

```diff
@@ -71,6 +78,39 @@ def list_enterprises(self):
     def list_domains(self):
         return self._request_list('/domains', 'get')
 
+    def list_l2_domains(self):
+        return self._request_list('/l2domains', 'get')
+
+    def list_floating_ips(self):
+        return self._request_list('/floatingips', 'get')
+
+    def list_vports_for_domain(self, domain_id):
+        return self._request_list('/domains/{}/vports'.format(domain_id), 'get')
+
+    def list_vports_for_l2_domain(self, l2_domain_id):
+        return self._request_list('/l2domains/{}/vports'.format(l2_domain_id), 'get')
+
+    def list_cloud_subnets(self):
+        return self.list_subnets() + self.list_l2_domains()
+
+    def list_vports(self):
+        res = []
+        for domain in self.list_domains():
+            res += self.list_vports_for_domain(domain['ID'])
```

Comment:
Please use `res.extend` instead of the `+=` operator for list extension.

Reply:
You could write these as comprehension as well:

`res = [self.list_vports_for_domain(domain['ID']) for domain in self.list_domains()]`

Reply:
Can't say no to one-liner that you suggested - got me hooked. Gimme 5min I'll come up with something. Problem is that list_vports_for_domain returns array so I'll need to flatten result. Thanks for suggestion.

Reply:
```python
def list_vports(self):
        return [self.list_vports_for_domain(d['ID']) for d in self.list_domains()] + \
            [self.list_vports_for_l2_domain(d['ID']) for d in self.list_l2_domains()]
```

@mshriver WDYT? ^

Reply:
So, if you want a list of lists, use append, and if you want one large list, use extend.

```
[1, 2, 3].append([4, 5])  == [1, 2, 3, [4, 5]]
[1, 2, 3].extend([4, 5]) == [1, 2, 3, 4, 5]

Reply:
Yeah but you can't do

```
return [1, 2, 3].append([4, 5])  == [1, 2, 3, [4, 5]]
```

Because it'll always return None. You need to do ether

```
arr = [1, 2, 3]
arr.extend([4, 5])
return arr
```

either

```
return [1, 2, 3] + [4, 5]
```

Which one you prefer?

Reply:
:) That was just meant as pseudo-code to show what extend/append will do with lists, you're correct that you need to assign that list to a name in order to properly extend/append it.